Test/HUnit/Lang.hs

```diff
@@ -89,4 +91,8 @@ performTestCase action =
        -- is not the case, please email the maintainer.
        E.Handler (\e -> throw (e :: E.AsyncException)),
 
+#if MIN_VERSION_base(4,9,0)
+       E.Handler (\(E.ErrorCallWithLocation msg _) -> return $ Error Nothing msg),
```

Comment:
There was a bit of a debate, using `String` was the pragmatic conclusion to avoid overengineering this (iirc -- I've forgotten the details). There' still room for improvement though if somebody comes up with a more desirable representation...

If you remove special casing at one end, you may need special casing in the testsuite instead... as you'd get varying `Show`-behaviour depending on the base-version used (some mangle the "error" string payload, some don't) ... :-)

There's no official 8.0RC1 yet, but @bgamari is working on getting RC1 out real-soon-now as we speak... 

However, https://github.com/hvr/multi-ghc-travis is already providing daily GHC 8.0 snapshots which are almost as good as the official RC1